tests/unit/s2n_random_test.c

```diff
@@ -793,25 +793,28 @@ static int s2n_random_rand_bytes_after_cleanup_cb(struct random_test_case *test_
 
 static int s2n_random_rand_bytes_before_init(struct random_test_case *test_case)
 {
-#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_RAND
-    /* Calling RAND_bytes will set a global random method */
-    unsigned char rndbytes[16] = { 0 };
-    EXPECT_EQUAL(RAND_bytes(rndbytes, sizeof(rndbytes)), 1);
-    const RAND_METHOD *rand_method = RAND_get_rand_method();
-    EXPECT_NOT_NULL(rand_method);
-    EXPECT_NOT_EQUAL(rand_method->bytes, s2n_openssl_compat_rand);
-
-    EXPECT_SUCCESS(s2n_init());
-
-    /* The global random method is overridden after calling s2n_init() */
-    const RAND_METHOD *custom_rand_method = RAND_get_rand_method();
-    EXPECT_NOT_NULL(custom_rand_method);
-    EXPECT_EQUAL(custom_rand_method->bytes, s2n_openssl_compat_rand);
+    if (s2n_supports_custom_rand()) {
+        /* Calling RAND_bytes will set a global random method */
+        unsigned char rndbytes[16] = { 0 };
+        EXPECT_EQUAL(RAND_bytes(rndbytes, sizeof(rndbytes)), 1);
+        const RAND_METHOD *rand_method = RAND_get_rand_method();
+        EXPECT_NOT_NULL(rand_method);
+#if S2N_LIBCRYPTO_SUPPORTS_ENGINE
+        EXPECT_NOT_EQUAL(rand_method->bytes, s2n_openssl_compat_rand);
+#endif
 
-    /* RAND_bytes is still successful */
-    EXPECT_EQUAL(RAND_bytes(rndbytes, sizeof(rndbytes)), 1);
+        EXPECT_SUCCESS(s2n_init());
 
+        /* The global random method is overridden after calling s2n_init() */
+        const RAND_METHOD *custom_rand_method = RAND_get_rand_method();
+        EXPECT_NOT_NULL(custom_rand_method);
+#if S2N_LIBCRYPTO_SUPPORTS_ENGINE
+        EXPECT_EQUAL(custom_rand_method->bytes, s2n_openssl_compat_rand);
 #endif
```

Comment:
Can s2n_openssl_compat_rand not compile without engine methods?

Reply:
It cant compile with AWS-LC due to the signature mismatch.

Reply:
Can cast to a void* to avoid compilation error

Reply:
casting to `void*` was resulted in a compilation error so I am instead casting to `typedef int (*s2n_rand_meth_fn_type)(unsigned char *buf, int num);`